pkg/rbac/common.go

```diff
@@ -237,14 +236,16 @@ func RulesFromTemplate(clusterRoles k8srbacv1.ClusterRoleCache, roleTemplates v3
 func gatherRules(clusterRoles k8srbacv1.ClusterRoleCache, roleTemplates v32.RoleTemplateCache, rt *v3.RoleTemplate, rules []rbacv1.PolicyRule, seen map[string]bool) ([]rbacv1.PolicyRule, error) {
 	seen[rt.Name] = true
 
-	if features.ExternalRules.Enabled() && rt.ExternalRules != nil {
-		rules = append(rules, rt.ExternalRules...)
-	} else if rt.External && rt.Context == "cluster" {
-		cr, err := clusterRoles.Get(rt.Name)
-		if err != nil {
-			return nil, err
+	if rt.External {
```

Comment:
I am not convinced that this change is correct. The original structure is

```
	if features.ExternalRules.Enabled() && rt.ExternalRules != nil {
		rules = append(rules, rt.ExternalRules...)
	} else if rt.External && rt.Context == "cluster" {
		cr, err := clusterRoles.Get(rt.Name)
		if err != nil {
			return nil, err
                }
		rules = append(rules, cr.Rules...)
	}
```

replacing `features.ExternalRules.Enabled()` with `true` yields

```
	if rt.ExternalRules != nil {
		rules = append(rules, rt.ExternalRules...)
	} else if rt.External && rt.Context == "cluster" {
		cr, err := clusterRoles.Get(rt.Name)
		if err != nil {
			return nil, err
                }
		rules = append(rules, cr.Rules...)
	}
```

I do not see how the `rt.External` clause can now be moved outside to become

```
if rt.External {
                if rt.ExternalRules != nil {
			rules = append(rules, rt.ExternalRules...)
		} else if rt.Context == "cluster" {
			cr, err := clusterRoles.Get(rt.Name)
			if err != nil {
				return nil, err
			}
			rules = append(rules, cr.Rules...)
		}
}
```

Reply:
`rt.ExternalRules != nil` implies `external` is true as the webhook enforces that a `RoleTemplate` can't have `ExternalRules` if `external` is not true. see https://github.com/rancher/webhook/blob/release/v0.5/pkg/resources/management.cattle.io/v3/roletemplate/validator.go#L132

That means that:
```
	if rt.ExternalRules != nil {
		rules = append(rules, rt.ExternalRules...)
	} else if rt.External && rt.Context == "cluster" {
		cr, err := clusterRoles.Get(rt.Name)
		if err != nil {
			return nil, err
                }
		rules = append(rules, cr.Rules...)
	}
```
and 

```
if rt.External {
                if rt.ExternalRules != nil {
			rules = append(rules, rt.ExternalRules...)
		} else if rt.Context == "cluster" {
			cr, err := clusterRoles.Get(rt.Name)
			if err != nil {
				return nil, err
			}
			rules = append(rules, cr.Rules...)
		}
}
```
behaves in the same way.  However, the second option is clearer and it'll work even if the webhook changes in the future

Reply:
Ok, the `rt.ExternalRules != nil implies external is true` is the part I had not known, and with that it makes sense.
Maybe add this as comment.